packages/story-editor/src/components/hotlinkModal/useHotlinkModal.js

```diff
@@ -107,6 +109,41 @@ function useHotlinkModal({
         return;
       }
 
+      if (
+        'image' === hotlinkInfo?.type &&
+        requiredImgDimensions?.height &&
+        requiredImgDimensions?.width
+      ) {
+        const proxiedUrl = needsProxy
+          ? getProxiedUrl({ needsProxy }, link)
+          : link;
+
+        const dimensions = await getImageDimensions(proxiedUrl);
+        if (
+          (requiredImgDimensions?.height &&
+            requiredImgDimensions?.height !== dimensions.height) ||
+          (requiredImgDimensions?.width &&
+            requiredImgDimensions?.width !== dimensions.width)
+        ) {
+          setErrorMsg(
+            sprintf(
+              /* translators: 1: supplied width. 2: supplied height. 3: desired width. 4: desired height */
+              __(
+                'Invalid image height supplied %1$d x %2$d when %3$d x %4$d is required.',
+                'web-stories'
+              ),
+              dimensions.width,
+              dimensions.height,
+              requiredImgDimensions.width,
+              requiredImgDimensions.height
+            )
```

Comment:
Let's use the following messaging for now:

If only a required **height** is provided:

```
Image height (%1$s) does not match required image height (%2$s)
```

Where the placeholders are the numbers including `px`. So it will be something like:

```
Image height (1234px) does not match required image height (853px)
```

If only a required **width** is provided:

```
Image width (%1$s) does not match required image width (%2$s)
```

If **both** required width **and** height are provided:

```
Image dimensions (%1$s) do not match required image dimensions (%2$s)
```

The "x" and the "px" of the dimensions are not part of the translatable string.

Example:

```
Image dimensions (1234x567px) do not match required image dimensions (640x853px)
```

Reply:
Fixed.